### PR TITLE
fix issue #27

### LINKED ISSUE #27: Volatiles stock are not parsed
A difference between doc and irl in Grocy.
Doc says it's a product schema but it's a CurrentStockReponse.
(Grocy API version : 2.4.4)
The fix is ready https://github.com/BlueBlueBlob/pygrocy/commit/2a2d04193f8447cb855a7432bd626e9fd5b5dcbe

### PR DESCRIPTION
**Pull request by BlueBlueBlob**
## Description

--Your description here--

**Related issue (if applicable):** fixes #27 

## Checklist
  - [x] The code change is tested and works locally.
  - [x] tests pass. **Your PR won't be merged unless tests pass**
  - [x] There is no commented out code in this PR

### CODE REVIEW

**Comment by BlueBlueBlob** [comment]
Sorry I thought I was able to valid pull request.
Must I do a direct push ?

**Comment by SebRut** [comment]
I have to fix the status checks for this repo, for now I have to manually merge your PRs, but ty for your work. If you want to communicate, I'm pretty much available on all platforms but most of the communication I've done for HA related stuff took place on Discord.

**Comment by BlueBlueBlob** [comment]
Ok nice and thanks.
I have Travis, codefactor and coverageall.io up and running. (Very useful btw)
All of my commits are what I did with Appdaemon and HA before seeing your lib 😹
And your HA custom_component can be awesome
Kind regards

File: pygrocy/grocy.py [hunk 1]
```diff
@@ -157,15 +157,39 @@ def stock(self, get_details: bool = False) -> List[Product]:
     def volatile_stock(self) -> CurrentVolatilStockResponse:
         return self._api_client.get_volatile_stock()
 
-    def expiring_products(self) -> List[ProductData]:
-        return self.volatile_stock().expiring_products
+    def expiring_products(self, get_details: bool = False) -> List[Product]:
+        raw_eg_product = self.volatile_stock().expiring_products
```
Comment (SebRut) [inline pygrocy/grocy.py]: Are `_eg_`, `_ei_`, `_m_  abbreviations for something I don't get? Please rename them to raw_expiring_products, raw_expired_products and raw_missing_products.

File: pygrocy/grocy_api_client.py [hunk 2]
```diff
@@ -168,31 +168,14 @@ def next_estimated_execution_time(self) -> datetime:
...
-        return self._missing_products
-
 
 class CurrentStockResponse(object):
     def __init__(self, parsed_json):
         self._product_id = parse_int(parsed_json.get('product_id'))
         self._amount = parse_float(parsed_json.get('amount'))
         self._best_before_date = parse_date(parsed_json.get('best_before_date'))
-
+        self._amount_opened = parse_int(parsed_json.get('amount_opened'))
```
Comment (SebRut) [inline pygrocy/grocy_api_client.py]: Is this really an int or should this be a float?

Comment (BlueBlueBlob) [reply pygrocy/grocy_api_client.py]: I don't know. Amount is a float. But I can not find a way to "open" 1.x
And documentation is not up to date :/

**Comment by BlueBlueBlob** [comment]
Not sure about what I did. Sorry, I'm learning rebase

**Comment by BlueBlueBlob** [comment]
Ok I messed up the pull request. And I will change to float (a float can do both)

**Pull request closed without merging**